ansible_galaxy/collection_info.py

```diff
@@ -24,9 +25,12 @@ def load(data_or_file_object, klass=None):
 
     log.debug('data: %s', pf(data_dict))
 
-    klass = CollectionInfo
-    collection_info = CollectionInfo(**data_dict)
-    collection_info = klass(**data_dict)
+    try:
+        collection_info = CollectionInfo(**data_dict)
+    except ValueError:
+        raise
+    except Exception as exc:
```

Comment:
I'd either (or maybe both):

1. avoid the hungry 'except Exception' and just let this bubble up for now. Could potentially obscure some exceptions we want to see and fix sooner than later, especially on py2 without chained exceptions.

2. log.exception(exc) here so we can troubleshoot it from logs.

Reply:
Letting it "bubble up" produces an error message with no context, leaving the user with no ability to fix the problem. And with a decent error, there's no need to log this.